src/components/RadioButtonGroup/RadioButtonGroup.tsx

```diff
@@ -49,25 +66,41 @@ const RadioButtonGroupComponent = (
     value: activeValue,
     name: customName,
     label,
+    invalid,
+    validationHint,
+    showValid,
+    disabled,
+    hasWarning,
     ...props
   }: RadioButtonGroupProps,
   ref: RadioButtonGroupProps['ref'],
 ) => {
   const name = customName || uniqueId('radio-button-group_');
   return (
-    <div role="group" aria-label={label} ref={ref} {...props}>
-      {options &&
-        options.map(({ children, value, className, ...rest }) => (
-          <div key={value && value.toString()} className={className}>
-            <RadioButton
-              {...{ ...rest, value, name, onChange }}
-              checked={value === activeValue}
-            >
-              {children}
-            </RadioButton>
-          </div>
-        ))}
-    </div>
+    <fieldset name={name}>
+      {label && <legend>{label}</legend>}
```

Comment:
The styling of the legend should match other input labels, so it should use the `kilo` text size (you could use the `textKilo` style helper) and have a `bit` sized bottom margin.